pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -15,41 +12,33 @@ import (
 
 	authapi "github.com/openshift/origin/pkg/auth/api"
 	"github.com/openshift/origin/pkg/auth/authenticator"
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
 )
 
 // Options contains configuration for an Authenticator instance
 type Options struct {
 	// URL is a parsed RFC 2255 URL
-	URL LDAPURL
-	// Insecure specifies if TLS is required for the connection. If true, either an ldap://... URL or StartTLS must be supported by the server
-	Insecure bool
-	// TLSConfig holds the TLS options. Only used when Insecure=false
-	TLSConfig *tls.Config
+	URL ldaputil.LDAPURL
+	// ServerConfig holds information about the LDAP server
+	ServerConfig ldaputil.LDAPServerConfig
```

Comment:
Looks to me like this refactor (and what it depends on) is logically distinct from your other changes.  Can you refactor the commits and pull this into a pull by itself?  I think I like it and I think we'll have a lot fewer comments than in the main body of this pull.